newIDE/app/src/Debugger/Inspectors/VariablesContainerInspector.js

```diff
@@ -37,27 +57,103 @@ const transform = variablesContainer => {
-  if (parseFloat(edit.new_value).toString() === edit.new_value) {
-    path.push('setNumber');
-    onCall(path, [parseFloat(edit.new_value)]);
-  } else {
-    path.push('setString');
-    onCall(path, ['' + edit.new_value]);
+/**
+ * Returns the list of properties to access the variable at the specified path in the specified variables container.
+ * Also returns the variable already living there.
+ */
+const constructPathToVariable = (
+  editPath: Array<string>,
+  variablesContainer: VariablesContainer
+): {| path: ?Array<string>, variable: ?Variable |} => {
+  const variableInContainerName = editPath.shift();
+  const path = ['_variables', 'items', variableInContainerName];
+  let variable = variablesContainer._variables.items[variableInContainerName];
+  let skip = false;
+
+  for (const variableName of editPath) {
+    // Skip every second key as it is the "value"
+    // key which is displayed only for better user
+    // experience but doesn't really exists
+    skip = !skip;
+    if (skip) continue;
+
+    // Walk down in the children of the collection.
+    if (variable._type === 'structure') {
+      path.push('_children', variableName);
+      variable = variable._children[variableName];
+    } else if (variable._type === 'array') {
+      path.push('_childrenArray', variableName);
+      variable = variable._childrenArray[parseInt(variableName, 10)];
+    }
+    // Bad path: abort.
+    else return { path: null, variable: null };
+  }
+
+  return { path, variable };
+};
+
+const handleEdit = (edit, { onCall, onEdit, variablesContainer }: Props) => {
+  if (!variablesContainer) return;
+
+  // Reconstruct the variable to edit from the path
+  const { path, variable } = constructPathToVariable(
+    edit.namespace,
+    variablesContainer
+  );
+  if (!path) {
+    console.error('Invalid path passed to the debugger: ', edit);
+    return false;
+  }
+  if (!variable) {
+    console.error("Variable doesn't exist: ", edit);
+    return false;
+  }
+
+  if (edit.name === 'type') {
+    path.push('castTo');
+    if (
+      edit.new_value === 'string' ||
+      edit.new_value === 'number' ||
+      edit.new_value === 'boolean' ||
+      edit.new_value === 'structure' ||
+      edit.new_value === 'array'
+    ) {
+      if (edit.new_value !== variable._type) onCall(path, [edit.new_value]);
+    } else {
+      console.error('Invalid type name: ' + edit.new_value);
+      return false;
+    }
+  } else if (edit.name === 'value') {
+    // Validate data type
+    if (
+      (variable._type === 'string' && typeof edit.new_value !== 'string') ||
+      (variable._type === 'number' && typeof edit.new_value !== 'number') ||
+      (variable._type === 'boolean' && typeof edit.new_value !== 'boolean')
+    ) {
+      console.error(
+        `Invalid value type: expected value of type '${
+          variable._type
+        }' but received a value of type '${typeof edit.new_value}'.`
+      );
+      return false;
```

Comment:
So an "interesting" thing I found (not entirely related to what you did in this PR) is that if you enter a value that is valid for booleans but don't press the bottom "validate" button in the editor, it will return a error:

![image](https://user-images.githubusercontent.com/1280130/106195446-0dde3b00-61a8-11eb-9cd7-163611200813.png)

See if I click on the top "☑️" button (instead of the bottom one):
![image](https://user-images.githubusercontent.com/1280130/106195540-2d756380-61a8-11eb-9491-aaf81f1116b1.png)

That's "fair enough", because this is a string instead of a boolean. Let me try to now change the variable again. After validating, this time I get another error:
![image](https://user-images.githubusercontent.com/1280130/106195620-4da52280-61a8-11eb-841f-8960b398ba3d.png)

because `edit.namespace` is empty 🤷‍♂️ Unsure what happened. But we can do at least one thing: do a parseFloat (for number), a string cast (`'' + edit.new_value`) or a "smart" boolean cast (`typeof edit.new_value === 'boolean' ? edit.new_value : edit.new_value === 'true'`) so that we handle most cases without errors (and so without breaking for some reason the ReactJsonView)